### ISSUE TITLE
Реализовать скачивание всех данных

### ISSUE THREAD

**Issue by DvaMishkiLapa** [DvaMishkiLapa/VKArchiveDownloader#8]
- [x] #9
- [x] #10
- [x] #11
- [x] #12
- [x] #13
- [x] #14

**Comment by 2naive** [comment]
@DvaMishkiLapa Как скачать фото в оригинальном разрешении из альбомов профиля?
Изображения скачиваются с префиксом x_  и небольшим разрешением ~600x450

**Comment by DvaMishkiLapa** [comment]
@2naive Через сам ВК скачиваются изображения лучшего качества? Скрипт собирает лишь доступные ссылки, который отдает ВК, не я выбираю размер изображения.

У меня есть альбом с изображением, где фотография скачалась в 1215х2160.

**Comment by 2naive** [comment]
@DvaMishkiLapa в архиве ВК ссылки в html не всегда ведут на изображение в наибольшем разрешении:
![image](https://github.com/DvaMishkiLapa/VKArchiveDownloader/assets/1451033/23ff74da-4ffc-4e04-9f27-b48c2cd99a8f)
При этом на самой странице фотографии по кнопке Скачать оригинал:
![image](https://github.com/DvaMishkiLapa/VKArchiveDownloader/assets/1451033/7a865db7-5bab-4c34-b325-8ac19bead10d)

В первом случае префикс X, во втором Z.
Детальнее тут: https://dev.vk.com/ru/reference/objects/photo-sizes

Соответственно, либо оригинал нужно брать по API, либо парсить страницу с фотографией.

**Comment by DvaMishkiLapa** [comment]
Да, и правда. Поправлю это. Осталось наплодить себе фотографий в альбом -_-

**Comment by DvaMishkiLapa** [comment]
@2naive 
Я так понимаю, речь идет про это:

> Для фотографий, загруженных на сайт до 2012 года, значения `width` и `height` могут быть недоступны, в этом случае соответствующие поля содержат 0.

Я нашел несколько фото старше 2012 года, но, кажется, я не могу просто взять и заменить прификс на более высокий по разрешению. Если у вас есть ссылки на одно и то же изображение с разными префиксами, я был бы рад на них взглянуть (если, конечно, вы готовы к их публикации).

В любом случае, странно, если в архив будут закладываться ссылки на оригинал не наивысшего качества. Если ссылка на то же самое изображение будет отличатся не только префиксом, я вряд ли что-то смогу с этим сделать.

**Comment by 2naive** [comment]
@DvaMishkiLapa 
`Я так понимаю, речь идет про это:`
-- нет, это другой случай.
`не могу просто взять и заменить прификс на более высокий по разрешению`
-- да, нужно получать полный список разрешений через API или парсить страницу с "Открыть оригинал", как я написал ранее
`В любом случае, странно, если в архив будут закладываться ссылки на оригинал не наивысшего качества`
-- странно, но именно так и есть, более того, по API массив со ссылками отдается тоже вразнобой без сортировки, а введенный тип 'w' сломал логику сортировки их по алфавиту
`Если ссылка на то же самое изображение будет отличатся`
-- да, ссылка уникальна, но выше указано, как можно решить

**Comment by 2naive** [comment]
https://github.com/2naive/vk-download/tree/master#%D0%B8%D0%BD%D1%81%D1%82%D1%80%D1%83%D0%BC%D0%B5%D0%BD%D1%82%D1%8B

**Comment by Lemonbrush** [comment]
Старые фотографии скачиваются в хорошем разрешении. Вот вложение с 2011 года

![x_4a7b4058](https://github.com/DvaMishkiLapa/VKArchiveDownloader/assets/14632982/78367cce-7a34-409e-ac45-452efb93cbd8)

**Comment by 2naive** [comment]
> Старые фотографии скачиваются в хорошем разрешении. Вот вложение с 2011 года
> 
> ![x_4a7b4058](https://user-images.githubusercontent.com/14632982/275351874-78367cce-7a34-409e-ac45-452efb93cbd8.jpg)

Grep-ните архив на предмет ссылок на изображения вида "x_******.jpg", как в примере: https://github.com/DvaMishkiLapa/VKArchiveDownloader/issues/8#issuecomment-1734372226

Если в вашем архиве есть такие ссылки - такие изображения будут скачаны в "плохом" качестве.
Скрипт просто берет то, что указано в архиве.
![image](https://github.com/DvaMishkiLapa/VKArchiveDownloader/assets/1451033/1088384b-7fa6-445d-a31f-e106ba2ee0df)
